proxy/src/control/destination/background.rs

```diff
@@ -40,6 +40,10 @@ use timeout::Timeout;
 type DestinationServiceQuery<T> = Remote<PbUpdate, T>;
 type UpdateRx<T> = Receiver<PbUpdate, T>;
 
+/// Duration to wait before polling DNS again for a negative result.
+// TODO: should this be a config option?
+const NEGATIVE_RESULT_TTL: Duration = Duration::from_secs(5);
```

Comment:
Does Trust-DNS not implement the RFC for this? Ideally this should be handled by Trust-DNS.

Reply:
@briansmith I've opened bluejekyll/trust-dns#485, which adds the deadline as a field on `ResolveErrorKind::NoRecordsFound` (which is returned by Trust-DNS on a NXDOMAIN response); once that lands we can wire it through here and use that deadline instead.